src/app/shared/deactivate-device/deactivate-device.component.ts

```diff
@@ -0,0 +1,78 @@
+import { Component, Inject, OnInit } from '@angular/core'
+import { FormBuilder, FormGroup, Validators } from '@angular/forms'
+import { MAT_DIALOG_DATA } from '@angular/material/dialog'
+import { MatTabChangeEvent } from '@angular/material/tabs'
+import { environment } from 'src/environments/environment'
+import { VaultService } from './vault.service'
+
+@Component({
+  selector: 'app-deactivate-device',
+  templateUrl: './deactivate-device.component.html',
+  styleUrls: ['./deactivate-device.component.scss']
+})
+export class DeactivateDeviceComponent implements OnInit {
+  vaultForm: FormGroup
+  hasPasswordretrieved: boolean = false
+  rpcLinux: string = 'sudo ./rpc '
+  rpcDocker: string = 'sudo docker run --device=/dev/mei0 rpc:latest '
+  rpcWindows: string = 'rpc.exe '
+  deactivationUrl: string = ''
+  serverUrl: string = `-u ${environment.rpsServer.replace('http', 'ws').replace('/rps', '')}/activate -c`
+  passwordString: string = ''
+  isCopied: boolean = false
+  error: boolean = false
+  selectedPlatform: string = 'linux'
+  constructor (@Inject(MAT_DIALOG_DATA) public data: {id: string}, public fb: FormBuilder, private readonly vaultService: VaultService) {
+    this.vaultForm = fb.group({
+      vaultPass: [null, Validators.required]
+    })
+  }
+
+  ngOnInit (): void {
+    this.formDeactivationUrl()
+  }
+
+  formDeactivationUrl (): void {
+    switch (this.selectedPlatform) {
+      case 'linux':
+        this.deactivationUrl = this.rpcLinux
+        break
+      case 'windows':
+        this.deactivationUrl = this.rpcWindows
+        break
+      case 'docker':
+        this.deactivationUrl = this.rpcDocker
+        break
+    }
+    this.deactivationUrl += `${this.serverUrl}${this.passwordString}`
+  }
+
+  onSubmit (): void {
+    if (this.vaultForm.valid) {
+      const result: any = Object.assign({}, this.vaultForm.value)
+      this.vaultService.getPassword(this.data.id, result.vaultPass)
+        .subscribe(response => {
+          this.error = false
+          // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+          this.passwordString = ` '-t deactivate --password ${response.data.data.AMT_PASSWORD}'`
```

Comment:
cast `AMT_PASSWORD as string` and remove eslint disable

Reply:
Updated